api/v1alpha4/metal3remediationtemplate_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// Metal3RemediationTemplateSpec defines the desired state of Metal3RemediationTemplate
+type Metal3RemediationTemplateSpec struct {
+	Template Metal3RemediationTemplateResource `json:"template"`
```

Comment:
Why don't we need to have `Metal3RemediationTemplateResource` instead of just using Metal3RemediationSpec here?

Reply:
In general I think every CRD needs to have it own `spec` and `status` even I does not include any fields, otherwise CRD can not be created. `Metal3RemediationTemplateSpec` is CRD's own `spec`. `Metal3RemediationTemplateResource` includes `Metal3RemediationSpec` as part of the `Metal3RemediationTemplateSpec`. Here is an example:

```
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha5
kind: Metal3RemediationTemplate
metadata:
    name: remediation-request
    namespace: metal3
spec: <-- Metal3RemediationTemplateSpec
  template: <-- Metal3RemediationTemplateResource 
    spec: <-- Metal3RemediationSpec
      strategy:
        type: "Reboot"
        retryLimit: 2
        timeout: 300s
```
This is typical way of implementing templating with CRDs. I hope this answer to you question.

Reply:
I see. It is clear to me now. Thanks for your answer, Jan